resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -373,20 +377,17 @@ private[spark] class Client(
       .map(CompressionUtils.createTarGzip(_))
   }
 
-  private def getDriverLauncherService(
-      k8ClientConfig: Config,
-      kubernetesMaster: String): KubernetesSparkRestApi = {
+  private def getDriverLauncherService(k8ClientConfig: Config): KubernetesSparkRestApi = {
     val url = s"${
       Array[String](
-        kubernetesMaster,
+        k8ClientConfig.getMasterUrl(),
```

Comment:
I'd recommend waiting on #57 since we won't be using the API server proxy for long.